src/transaction.ts

```diff
@@ -29,24 +29,24 @@ import {PartialResultStream, partialResultStream, ResumeToken, Row} from './part
 import {Session} from './session';
 import {Key} from './table';
 import {SpannerClient as s} from './v1';
+import {RequestCallback, RowCountsServiceError, ITransactionOptions, ITransaction, GeneratedTransaction, ITimestamp, IReadOnly, ICommitResponse, IMutation} from './common';
 
 export type Rows = Array<Row|Json>;
-
-export interface TimestampBounds {
-  strong?: boolean;
-  minReadTimestamp?: PreciseDate|p.ITimestamp;
-  maxStaleness?: number|p.IDuration;
-  readTimestamp?: PreciseDate|p.ITimestamp;
-  exactStaleness?: number|p.IDuration;
-  returnReadTimestamp?: boolean;
+export type BeginResponse = [s.Transaction];
+export type CommitResponse = [ICommitResponse];
+export type ReadResponse = [Rows];
+export type RunResponse = [Rows, s.ResultSetStats];
+export type RunUpdateResponse = [number];
+
+export type BeginTransactionCallback = RequestCallback<GeneratedTransaction>;
+export type CommitCallback = RequestCallback<ICommitResponse>;
+
+export interface CommitRequest {
```

Comment:
Sorry if I asked this before, but why do we need this? Isn't there a `ICommitRequest` that covers it?

Reply:
`ICommitRequest` seems to, by default, annotate properties as `undefined` and `null`, so this was a solution that made it a bit stricter where needed. I'll play around with this one too